src/export/pdf-exporter.ts

```diff
@@ -1,19 +1,27 @@
 import { WebviewTag } from 'electron';
-import * as FileExtraUtils from 'fs-extra';
+import { Exporter, ExportResult } from './exporter';
 
-export class PdfExporter {
-	public static exportToPdf(path: string, webview: WebviewTag): void {
-		webview.printToPDF(
-			{
-				marginsType: 1,
-				pageSize: 'A4',
-				printBackground: true,
-				printSelectionOnly: false,
-				landscape: false
-			},
-			(error, pdfBuffer) => {
-				FileExtraUtils.writeFileSync(path, pdfBuffer);
-			}
-		);
+export class PdfExporter extends Exporter {
+	// tslint:disable-next-line promise-function-async
```

Comment:
It would be great if we can remove this one. Or at least have a comment next to it why it is needed

Reply:
`promise-function-async` forbids `Promise<any>` returning functions that are not marked `async`. That's nonsensical, because it prevents use from building bridges between classical `(err: null | Error, result?: any) => void` Node.js APIs (like `printToPdf`) and `Promsise<any>` based control flow. 

I'd like to disable the rule in a separate PR instead of explaining the comment here.

Reply:
Did the full typing fashizzle required when using `async`